data_generator.py

```diff
@@ -111,6 +121,49 @@ def generate_heatmaps(self, label):
 
         return heat_maps
 
+    def convert_coco_kp_to_imgaug_kp(self, label):
+        kps = []
+        valid = np.ones(NUM_COCO_KEYPOINTS)
+        invalid_xy = -1
+        for i in range(NUM_COCO_KEYPOINTS):
+            label_idx = i * NUM_COCO_KP_ATTRBS  # index for label
+            # generate empty heatmap for unlabelled kp
+            if label[label_idx + (NUM_COCO_KP_ATTRBS-1)] == 0:
+                # invalid keypoint
+                valid[i] = 0
+                kps.append(Keypoint(x=invalid_xy, y=invalid_xy))
+                continue
+            kpx = int(label[label_idx])
+            kpy = int(label[label_idx + 1])
+            kps.append(Keypoint(x=kpx, y=kpy))
+
+        return kps, valid
+
+    def convert_imgaug_kpsoi_to_coco_kp(self, kpsoi_aug, valid, image_aug):
+        transformed_label = []
+
+        # Keypoint is visible
+        VALID_KP_V = 1
+        for i in range(NUM_COCO_KEYPOINTS):
+            kp = kpsoi_aug[i]
+            if (not valid[i]) or kp.is_out_of_image(image_aug):
+                x, y, v = (0, 0, 0)
+            else:
+                x = kp.x
+                y = kp.y
+                v = VALID_KP_V
+                ## NOTE: Uncomment below if we wish to keep visibility 1,2 flag
+                # label_idx = i * NUM_COCO_KP_ATTRBS  # index for label
+                # visibility = label[label_idx + (NUM_COCO_KP_ATTRBS-1)]
+                # v = visibility
```

Comment:
in `transform_label` we currently keep v=0 and turn 1 and 2 into 1. So this comment would need to be applied to the previous method

Reply:
Ah, I'll remove then.